File: backend/app/models/course.py
from sqlalchemy import Boolean, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import BaseModel


class Course(BaseModel):
    __tablename__ = "courses"

    language_id: Mapped[str] = mapped_column(
        ForeignKey("languages.id"),
        nullable=False,
        index=True,
    )

    level: Mapped[str] = mapped_column(
        String(20),
        nullable=False,
    )

    title: Mapped[str] = mapped_column(
        String(255),
        nullable=False,
    )

    description: Mapped[str | None] = mapped_column(
        String(1000),
        nullable=True,
    )

    order_index: Mapped[int] = mapped_column(
        Integer,
        default=1,
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        default=True,
    )

    language = relationship(
        "Language",
        back_populates="courses",
    )

    modules = relationship(
        "Module",
        back_populates="course",
        cascade="all, delete-orphan",
        order_by="Module.order_index",
    )

File: frontend/src/features/admin/types/course.ts
export interface AdminCourseItem {
  id: string;
  languageId: string;
  level: string;
  title: string;
  description: string | null;
  orderIndex: number;
  isActive: boolean;
}

export interface CreateCourseInput {
  languageId: string;
  level: string;
  title: string;
  description?: string;
  orderIndex?: number;
  isActive?: boolean;
}

export interface UpdateCourseInput {
  languageId?: string;
  level?: string;
  title?: string;
  description?: string;
  orderIndex?: number;
  isActive?: boolean;
}
